tasks/release.rake

```diff
@@ -1,6 +1,42 @@
 require "chandler/tasks"
 
+root    = File.expand_path("..", __dir__)
+version = File.read("#{root}/VERSION").strip
+
+# See https://github.com/rails/rails/blob/master/tasks/release.rb#L26
```

Comment:
This link is now outdated, we should use a permalink if linking to Rails code. In any case, the ruby-like version to npm-like version is incorrect like they point out in the source code comment.

I think we can use a more correct version of the conversion like

```ruby
def npmify(version)
  segments = version.split(".")

  split_point = segments.index{|segment| segment.to_s =~ /[a-z]/}

  release_segments, prerelease_segments = segments[0...split_point], segments[split_point..-1]

  [release_segments.join("."), prerelease_segments.join(".")].join("-")
end
```

which gives

```
irb(main):016:0> npmify "5.0.1"
=> "5.0.1"
irb(main):017:0> npmify "5.0.1.1"
=> "5.0.1.1"
irb(main):018:0> npmify "5.0.0.rc1"
=> "5.0.0-rc1"
```

Reply:
Actually, scratch that. Things like "5.1.1.1" are invalid according to semver since semver doesn't define a TINY (4th) segment. So it's undefined what we should do with that, and we don't use that numbering in activeadmin anyways.

So let's go with the `Rails` version. Only thing I would tweak is to only convert the dot separating the prerelease part from the release part to a hyphen, because the result looks less weird and more similar to the "ruby version". So `==` instead of `>=`:

```
npm_version = version.gsub(/\./).with_index { |s, i| i == 2 ? "-" : s } 
```

Reply:
As [commented in rails repo](https://github.com/rails/rails/blob/0d0c30e534af7f80ec8b18eb946aaa613ca30444/tasks/release.rb#L27) versions like `5.0.0.beta1.1` should be transformed to `5.0.0-beta1-1`, so I guess we should keep `>=` but I'm no semver expert and haven't seen any such version so far so it's really up to you

Reply:
Yeah, that's the case I want to change. `5.0.0-beta1-1` looks very weird to me, and `5.0.0-beta1.1` is perfectly semver compliant and less ambiguos in my opinion since it has a single hyphen separating the release part from the prerelease part.